kernel/src/process_standard.rs

```diff
@@ -1385,14 +1386,46 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
                         "Process not enabled flash={:#010X}-{:#010X} process={:?}",
                         app_flash.as_ptr() as usize,
                         app_flash.as_ptr() as usize + app_flash.len() - 1,
-                        process_name
+                        process_name.unwrap_or("(no name)")
                     );
                 }
             }
             // Return no process and the full memory slice we were given.
             return Ok((None, remaining_memory));
         }
 
+        if let Some((major, minor)) = tbf_header.get_kernel_version() {
+            // If the `KernelVersion` header is present, we read the requested kernel version and compare it to
+            // the running kernel version.
+            if crate::MAJOR != major {
+                // If the kernel major version is different, we prevent the process from being loaded.
+                if config::CONFIG.debug_load_processes {
+                    debug!("WARN process {:?} not loaded as it requires kernel version {} (running kernel {}.{})", process_name.unwrap_or("(no name)"), major, crate::MAJOR, crate::MINOR);
+                }
+                return Ok((None, remaining_memory));
+            } else if crate::MINOR < minor {
+                // If the kernel major version is the same, we compare the kernel minor version. The current
+                // running kernel minor version has to be greater or equal to the one that the process
+                // has requested. If not, we prevent the process from loading.
+                if config::CONFIG.debug_load_processes {
+                    debug!("WARN process {:?} not loaded as it requires kernel version at least {}.{}, (running kernel {}.{})", process_name.unwrap_or("(no name)"), major, minor, crate::MAJOR, crate::MINOR);
```

Comment:
Do we need to separate these cases?

Reply:
If we want to display different errors, but I suppose we could merge this into a single if.